sklearn/model_selection/_validation.py

```diff
@@ -639,9 +639,20 @@ def cross_val_predict(estimator, X, y=None, groups=None, cv=None, n_jobs=1,
 
     cv = check_cv(cv, y, classifier=is_classifier(estimator))
 
-    if method in ['decision_function', 'predict_proba', 'predict_log_proba']:
-        le = LabelEncoder()
-        y = le.fit_transform(y)
+    # If classification methods produce multiple columns of output,
+    # we need to manually encode classes to ensure consistent column ordering.
+    encode = method in ['decision_function', 'predict_proba',
+                        'predict_log_proba']
+    if encode:
+        y = np.asarray(y)
+        if y.ndim == 1:
+            le = LabelEncoder()
+            y = le.fit_transform(y)
+        elif y.ndim == 2:
+            y_enc = np.zeros_like(y, dtype=np.int)
+            for i_label in range(y.shape[1]):
```

Comment:
This seems redundant in the multi-label case (not multioutput multiclass)

Reply:
Yes, it is. The same redundancy is present in the original 1D case, for what it's worth. The `y` gets encoded and transformed whether it needs it or not. It does seem more costly in the 2D case to run the encoding when it's not needed. 

The only way I'm thinking to check whether we're in a multi-label case is to run a `set(y.flat) == {0, 1}` check, or maybe `np.unique(y).tolist() == [0, 1]`. Do you think the performance tradeoff is worth it?